test/e2e/tests/test_pipeline.py

```diff
@@ -0,0 +1,170 @@
+        pipeline_spec,
+        pipeline_resource,
+    )
+
+    # Delete the k8s resource if not already deleted by tests
+    if k8s.get_resource_exists(pipeline_reference):
+        _, deleted = k8s.delete_custom_resource(
+            pipeline_reference, DELETE_WAIT_PERIOD, DELETE_WAIT_LENGTH
+        )
+        assert deleted
+
+
+def get_sagemaker_pipeline_status(pipeline_arn: str):
+    sm_pipeline_desc = get_sagemaker_pipeline(pipeline_arn)
+    return sm_pipeline_desc["PipelineStatus"]
+
+
+def get_pipeline_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    assert "pipelineStatus" in resource["status"]
+    return resource["status"]["pipelineStatus"]
+
+
+@pytest.mark.canary
+@service_marker
+class TestPipeline:
+    def _wait_resource_pipeline_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_pipeline_resource_status,
+            reference,
+        )
+
+    def _wait_sagemaker_pipeline_status(
+        self,
+        pipeline_arn,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_sagemaker_pipeline_status,
+            pipeline_arn,
+        )
+
+    def _assert_pipeline_status_in_sync(self, pipeline_arn, reference, expected_status):
+        assert (
+            self._wait_sagemaker_pipeline_status(pipeline_arn, expected_status)
+            == self._wait_resource_pipeline_status(reference, expected_status)
+            == expected_status
+        )
+
+    def test_pipeline_succeeded(self, pipeline):
+        (reference, spec, resource) = pipeline
+        assert k8s.get_resource_exists(reference)
+
+        pipeline_name = resource["spec"].get("pipelineName")
+        # Need PipelineArn to reference the resource
+
+        pipeline_desc = get_sagemaker_pipeline(pipeline_name=pipeline_name)
+        if k8s.get_resource_arn(resource) is None:
+            logging.error(
+                f"ARN for this resource is None, resource status is: {resource['status']}"
+            )
+
+        pipeline_arn = pipeline_desc["PipelineArn"]
+        old_pipeline_last_modified_time = pipeline_desc["LastModifiedTime"]
+        assert k8s.get_resource_arn(resource) == pipeline_arn
+
+        self._assert_pipeline_status_in_sync(pipeline_arn, reference, "Active")
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
+
+        # Update the resource
+        new_pipeline_display_name = random_suffix_name("updated-display-name", 38)
+        spec["spec"]["pipelineDisplayName"] = new_pipeline_display_name
```

Comment:
can you also update the pipeline definition? a better use case

Reply:
We can choose to update the pipeline definition, would just be a long string in the file., if thats cool for the e2e test then sure.

Reply:
sure, lets do in followup PR